src/Sfx/App/Scripts/Models/DataModels/Cluster.ts

```diff
@@ -40,6 +65,61 @@ module Sfx {
             return this.emptyHealthStateCount;
         }
 
+        private setMessage(healthEvent: HealthEvent): void {
+            /*
+            Example description for parsing reference(if this message changes this might need updating)
+            Certificate expiration: thumbprint = 35d8f6bb4c52bd3f40a327a3094a9ee9692679ce, expiration = 2020-03-13 22:23:40.000
+            , remaining lifetime is 213:8:17:08.174, please refresh ahead of time to avoid catastrophic failure.
+            Warning threshold Security/CertificateExpirySafetyMargin is configured at 289:8:26:40.000, if needed, you can
+            adjust it to fit your refresh process.
+
+            */
+
+            const thumbprintSearchText = "thumbprint = ";
+            const thumbprintIndex = healthEvent.raw.Description.indexOf(thumbprintSearchText);
+            const thumbprint =  healthEvent.raw.Description.substr(thumbprintIndex + thumbprintSearchText.length).split(",")[0];
+
+            const expirationSearchText = "expiration = ";
+            const expirationIndex = healthEvent.raw.Description.indexOf("expiration = ");
+            const expiration = healthEvent.raw.Description.substring(expirationIndex + expirationSearchText.length).split(",")[0];
+
+            this.data.warnings.addOrUpdateNotification({
+                message: `A cluster certificate is set to expire soon. Replace it as soon as possible to avoid catastrophic failure. <br> Thumbprint : ${thumbprint}    Expiration: ${expiration}`,
+                level: StatusWarningLevel.Error,
+                priority: 5,
+                id: BannerWarningID.ExpiringClusterCert,
+                link: "https://aka.ms/sfrenewclustercert/",
+                linkText: "Read here for more guidance"
+            });
+            ClusterHealth.certExpirationChecked = true;
+        }
+
+        private containsCertExpiringHealthEvent(unhealthyEvaluations: HealthEvent[]): HealthEvent[] {
+            return unhealthyEvaluations.filter(event => event.raw.Description.indexOf("Certificate expiration") === 0 &&
+                                             event.raw.Property === CertExpiraryHealthEventProperty.Cluster &&
```

Comment:
Realistically, this won't be a problem, but curious as to why sourceId is not also compared?

Reply:
I wouldve missed that, I can add it if it would make a difference?

Reply:
It is safer to add that check, but in practice I doubt you will run into any problems by not adding it